### PR TITLE
[msl] namespace and matrix fixes to get Skybox running

### PR DESCRIPTION
**Pull request by kvark**
Based on #329, this PR gets us to a point where we can generate the MSL shader for the `skybox` example directly from IR 🎉 
(there is only one new commit here)
The things missing were:
  - inserting the namespace qualifier `metal::` before the types (we can discuss this on Matrix if there are concerns)
  - fixing matrix composition and indexing

Update: also got GLSL output working for this! And added it to the snapshot tests.

### CODE REVIEW

**Review by JCapucho — CHANGES_REQUESTED** [review]
Reviewing the glsl backend, there's just one small thing that should be fixed (was wrong before this PR), everything else looks fine to me

File: src/back/glsl/features.rs [hunk 1]
```diff
@@ -193,7 +193,7 @@ impl<'a, W> Writer<'a, W> {
             match ty.inner {
                 TypeInner::Scalar { kind, width } => self.scalar_required_features(kind, width),
                 TypeInner::Vector { kind, width, .. } => self.scalar_required_features(kind, width),
-                TypeInner::Matrix { .. } => self.scalar_required_features(ScalarKind::Float, 8),
+                TypeInner::Matrix { .. } => self.scalar_required_features(ScalarKind::Float, 4),
```
Comment (JCapucho) [inline src/back/glsl/features.rs:196]: This should be
```rust
TypeInner::Matrix {width, .. } => self.scalar_required_features(ScalarKind::Float, width),
```

**Review by Napokue — APPROVED** [review]
Reviewed everything outside the GLSL back-end. LGTM!

File: src/back/glsl/features.rs (hunk 1 above)
Comment (kvark) [reply src/back/glsl/features.rs:196]: uuh, how did I miss that :) thanks for spotting!

**Pull request merged**